maas/client/viscera/tests/test_ip_addresses.py

```diff
@@ -0,0 +1,29 @@
+"""Tests for `maas.client.viscera.ip_addresses`."""
+
+from testtools.matchers import Equals
+
+from .. import ip_addresses
+
+from ..testing import bind
+from ...testing import TestCase
+
+
+def make_origin():
+    """
+    Create a new origin with IPAddress and IPAddresses. The former
+    refers to the latter via the origin, hence why it must be bound.
+    """
+    return bind(ip_addresses.IPAddresses, ip_addresses.IPAddress)
+
+
+class TestIPAddresses(TestCase):
+    def test__ip_addresses_read(self):
+        """IPAddresses.read() returns a list of IPAddresses."""
+        IPAddresses = make_origin().IPAddresses
+        ip_addresses = [
+            {"ip": "10.0.0.%s" % (i + 1), "alloc_type_name": "User reserved"}
```

Comment:
is the version of python we're using too old to support f-strings?
ie: `f"10.0.0.{i+1}"` rather than `"10.0.0.%s" % (i+1)`

Reply:
I searched for usage of f-strings and there is none present at this repo. I decided to use the existing pattern so as to not differentiate. If you feel it is ok as is, please resolve and I will merge afterwards

Reply:
That's fair, may as well stick with the established code style